lib/qs/client.rb

```diff
@@ -66,6 +67,39 @@ def ping
         self.redis.with{ |c| c.ping }
       end
 
+      def read_subscriptions(queue)
+        self.redis.with{ |c| c.smembers(queue.subscriptions_redis_key) }
+      end
+
+      def sync_subscriptions(queue)
+        event_subs_keys = queue.event_job_names.map do |event_job_name|
+          Qs::Event::SubscribersRedisKey.new(event_job_name)
+        end
+        redis_transaction do |c|
+          c.hmset(queue.pubsub_redis_key, 'version', Qs::VERSION)
+          c.del(queue.subscriptions_redis_key)
+          c.sadd(queue.subscriptions_redis_key, *queue.event_job_names)
+          event_subs_keys.each{ |key| c.sadd(key, queue.name) }
+        end
+      end
+
+      def clear_subscriptions(queue)
+        event_subs_keys = self.read_subscriptions(queue).map do |job_name|
+          Qs::Event::SubscribersRedisKey.new(job_name)
```

Comment:
@jcredding minor, but can we call this local `event_job_name` here to match what it is called in `sync_subscriptions` above?